src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -55,64 +58,116 @@
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
-    throws ServletException, IOException {
-    String stringToMatch = request.getParameter("searchterm");
-    stringToMatch = trimAndRemoveCommas(stringToMatch);
-    request.setAttribute("searchQuery", stringToMatch);
-
-    List<Recipe> recipesListToReturn = recipesMatching(
-      request,
-      response,
-      stringToMatch
+    throws ServletException, IOException, NullPointerException {
+    String searchTerm = getParameter("searchterm", request);
+    String difficulty = getParameter("difficulty", request);
+    String time = getParameter("time", request);
+
+    request.setAttribute("searchTerm", searchTerm);
+    request.setAttribute("difficulty", difficulty);
+    request.setAttribute("prepTime", time);
+
+    searchTerm = sanitizeString(searchTerm);
+    difficulty = sanitizeString(difficulty);
+    request.setAttribute(
+      "difficulty",
+      difficulty
     );
 
-    request.setAttribute("recipesList", recipesListToReturn);
+
+    Integer timeValue;
+    try {
+      timeValue = Integer.valueOf(sanitizeString(time));
+    } catch (NumberFormatException e) {
+      timeValue = 0;
+    }
+
+    try {
+      List<Recipe> recipesListToReturn = recipesMatching(
+        request,
+        response,
+        searchTerm,
+        difficulty,
+        timeValue
+      );
+      request.setAttribute("recipesList", recipesListToReturn);
+    } catch (SearchQueryException e) {
+      response.setStatus(505);
+    }
     request.getRequestDispatcher("/search.jsp").forward(request, response);
   }
 
   /**
    * Replace and return the string with no commas and no multiple consecutive or trailing spaces.
    */
-  private String trimAndRemoveCommas(String stringToRemove) {
-    if (!StringUtils.isBlank(stringToRemove)) {
-      stringToRemove = stringToRemove.replaceAll("[,]+", " ");
-      stringToRemove = stringToRemove.replaceAll("  +", " ");
+  private String sanitizeString(String stringToSanitize) {
+    if (!StringUtils.isBlank(stringToSanitize)) {
+      stringToSanitize = stringToSanitize.replaceAll("(,\\s?)+", " ");
+    }
+    return stringToSanitize;
+  }
+
+  /**
+   * If the fields exists, returns the parameter from the request, otherwise it returns an empty string.
+   */
+  private String getParameter(String field, HttpServletRequest request) {
+    String parameter = new String();
+    try {
+      parameter = request.getParameter(field);
+    } catch (NullPointerException e) {
```

Comment:
Instead of catching a null pointer exception you could check with an if directly

if (field != null) {
return request.getParameter(field);
}

return "";